src/main/java/gregtech/loaders/recipe/RecyclingRecipes.java

```diff
@@ -1,111 +1,400 @@
+                .duration((int) duration)
+                .EUt(32 * multiplier);
+
+        // Null check the Item before adding it to the Builder.
+        // - Try to output an Ingot, otherwise output a Dust.
+        if (itemMs != null) {
+            OrePrefix prefix = itemMs.material.hasProperty(PropertyKey.INGOT) ? OrePrefix.ingot : OrePrefix.dust;
+            extractorBuilder.output(prefix, itemMs.material, (int) (itemMs.amount / M));
         }
+
+        extractorBuilder.buildAndRegister();
+    }
+
+    private static void registerArcRecycling(ItemStack input, List<MaterialStack> materials, int multiplier) {
+
+        // Filter down the materials list.
+        // - Map to the Arc Smelting result as defined below
+        // - Combine any MaterialStacks that have the same Material
+        materials = combineStacks(materials.stream()
+                .map(RecyclingRecipes::getArcSmeltingResult)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList()));
+
+        // Finalize the output List
+        List<ItemStack> outputs = finalizeOutputs(
+                materials,
+                RecipeMaps.ARC_FURNACE_RECIPES.getMaxOutputs(),
+                RecyclingRecipes::getArcIngotOrDust
+        );
+
+        // Exit if no valid outputs exist for this recycling Recipe.
+        if (outputs.size() == 0) return;
+
+        // Build the final Recipe.
+        RecipeMaps.ARC_FURNACE_RECIPES.recipeBuilder()
+                .inputs(input)
+                .outputs(outputs)
+                .duration(calculateDuration(materials))
+                .EUt(30 * multiplier)
+                .buildAndRegister();
     }
 
-    private static ItemStack getArcSmeltingResult(MaterialStack materialStack) {
+    private static MaterialStack getArcSmeltingResult(MaterialStack materialStack) {
         Material material = materialStack.material;
-        long materialAmount = materialStack.amount;
+        long amount = materialStack.amount;
+
+        if (material.hasFlag(EXPLOSIVE)) {
+            return new MaterialStack(Materials.Ash, amount / 16);
+        }
+
+        // If the Material is Flammable, return Ash
         if (material.hasFlag(FLAMMABLE)) {
-            return OreDictUnifier.getDust(Materials.Ash, materialAmount);
-        } else if (material.hasProperty(PropertyKey.GEM)) {
-            if (materialStack.material.getMaterialComponents().stream()
-                    .anyMatch(stack -> stack.material == Materials.Oxygen)) {
-                return OreDictUnifier.getDust(Materials.Ash, materialAmount);
+            return new MaterialStack(Materials.Ash, amount / 8);
+        }
+
+        // Else if the Material is a Gem, process its output (see below)
+        if (material.hasProperty(PropertyKey.GEM)) {
+            return getGemArcSmeltResult(materialStack);
+        }
+
+        // Else if the Material has NO_SMELTING, return nothing
+        if (material.hasFlag(NO_SMELTING)) {
```

Comment:
This prevents things like Brick Dust and Stone Dust from being output in the process